tephra-hbase-compat-1.1-base/src/main/java/org/apache/tephra/hbase/coprocessor/TransactionProcessor.java

```diff
@@ -303,10 +305,22 @@ public InternalScanner preCompactScannerOpen(ObserverContext<RegionCoprocessorEn
     // Get the latest tx snapshot state for the compaction
     TransactionVisibilityState snapshot = cache.getLatestState();
 
-    // Record tx state before the compaction
-    if (compactionState != null) {
+    if (pruneEnable == null) {
+      Configuration conf = getConfiguration(c.getEnvironment());
+      pruneEnable = conf.getBoolean(TxConstants.TransactionPruning.PRUNE_ENABLE,
```

Comment:
If the configuration is not available for the first time then won't this become false for the whole run? Should we return null from `getConfiguration` in case configuration is not available? This way we can skip the run